### ISSUE TITLE
problem compiling on archlinux

### ISSUE THREAD

**Issue by pietrondo** [reicast/reicast-emulator#896]
Problem compiling with archlinux 

```
mkdir -p obj-dreamcast-x64-sse4_1/rend/soft/
g++   -I../../core -I../../core/deps -I../../core/khronos  -D TARGET_LINUX_x64 -D TARGET_NO_AREC -fsingle-precision-constant -g -O3 -D RELEASE -c -D USES_HOMEDIR  -frename-registers -fno-strict-aliasing -ffast-math -ftree-vectorize -fomit-frame-pointer -D SUPPORT_X11 -fopenmp -msse4.1   -fexceptions -fno-rtti -fpermissive -fno-operator-names -D SUPPORT_X11  -D USE_EVDEV -D USE_JOYSTICK -D USE_ALSA -D USE_OSS -DTARGET_SOFTREND ../../core/rend/soft/softrend.cpp -o obj-dreamcast-x64-sse4_1/rend/soft/softrend.build_obj
../../core/rend/soft/softrend.cpp: In member function ‘__m128 softrend::shuffle_pixel(__m128)’:
../../core/rend/soft/softrend.cpp:1167:77: error: invalid cast of an rvalue expression of type ‘__m128i {aka __vector(2) long long int}’ to type ‘__m128& {aka __vector(4) float&}’
   return (__m128&)_mm_shuffle_epi8((__m128i&)v, _mm_set_epi8(R(0x80,2,1, 0)));

```

gcc version:

```
gcc version 5.3.0 (GCC) 
```

**Comment by gamax92** [comment]
Is this on the ```fix/softrend-fugly-casts``` branch? if not try that next.

**Comment by bmaupin** [comment]
I have a hunch this was fixed by #914.

**Comment by skmp** [comment]
I'll close this then, and we can open a new issue if the problem persists.
